scipy/signal/tests/test_signaltools.py

```diff
@@ -2422,42 +2384,32 @@ def equal_tolerance_fft(self, res_dt):
 
     @skip_xp_backends(np_only=True, reason="order='F'")
     def test_method(self, dt, xp):
-        if dt == Decimal:
-            method = choose_conv_method([Decimal(4)], [Decimal(3)])
-            assert method == 'direct'
-        else:
-            dt = getattr(xp, dt)
+        dt = getattr(xp, dt)
 
-            a, b, y_r = self._setup_rank3(dt, xp)
-            y_fft = correlate(a, b, method='fft')
-            y_direct = correlate(a, b, method='direct')
+        a, b, y_r = self._setup_rank3(dt, xp)
+        y_fft = correlate(a, b, method='fft')
+        y_direct = correlate(a, b, method='direct')
 
-            assert_array_almost_equal(y_r,
-                                      y_fft,
-                                      decimal=self.equal_tolerance_fft(y_fft.dtype),)
-            assert_array_almost_equal(y_r,
-                                      y_direct,
-                                      decimal=self.equal_tolerance(y_direct.dtype),)
-            assert y_fft.dtype == dt
-            assert y_direct.dtype == dt
+        assert_array_almost_equal(y_r, y_fft,
+                                  decimal=self.equal_tolerance_fft(y_fft.dtype),)
+        assert_array_almost_equal(y_r, y_direct,
+                                  decimal=self.equal_tolerance(y_direct.dtype),)
+        assert y_fft.dtype == dt
+        assert y_direct.dtype == dt
 
     def test_rank1_valid(self, dt, xp):
         if is_torch(xp) and dt in ["uint16", "uint32", "uint64"]:
            pytest.skip("torch does not support unsigned ints")
 
         dt = getattr(xp, dt) if isinstance(dt, str) else dt
-        _assert_almost_equal = self._get_assertion(dt)
         a, b, y_r = self._setup_rank1(dt, xp)
         y = correlate(a, b, 'valid')
-        _assert_almost_equal(y, y_r[1:4])
+        assert_allclose(y, y_r[1:4])
```

Comment:
```suggestion
        assert_array_almost_equal(y, y_r[1:4])
```

which eventually calls `xp_assert_equal`